src/js/actions/tools.js

```diff
@@ -791,8 +789,24 @@ define(function (require, exports) {
      */
     var beforeStartup = function () {
         var flux = this.flux,
+            appStore = flux.store("application"),
             toolStore = this.flux.store("tool"),
-            tools = toolStore.getAllTools();
+            tools = toolStore.getAllTools(),
+            documentLayerBounds;
+
+        var throttledResetBorderPolicies =
+            synchronization.throttle(this.flux.actions.tools.resetBorderPolicies, this, 100),
+            _documentOrApplicationChangeHandler = function () {
+                var nextDocuemntLayerBounds = appStore.getCurrentDocument().layers.selectedChildBounds;
```

Comment:
If there is no current document, this will throw an unhandled exception. Also, there's a typo in the variable name.